### DISCUSSION TITLE
[Share] Add tag shortcut (working in Zotero 7)

### DISCUSSION THREAD

**Discussion by powestermark** [windingwind/zotero-actions-tags#390] [Action Scripts]
### Description

A script suitable for a keyboard shortcut for adding a tag. 

Very bare bones, found through some experimenting and with help from ChatGPT. Fails silently if pane is not visible (querySelector() returns null).

(previous solution https://github.com/windingwind/zotero-actions-tags/discussions/139 does not work in Zotero 7).

**Edit:** Updated to work also in the pdf reader!

**Edit:** Does NOT work in Zotero 8. Se #550 for my new script that works in Zotero 8. 

### Event

None

### Operation

Script

### Data

```javascript
/**
 * Add tag (if pane is visible).
 * @author Zotero Community
 * @usage 
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/
 */

// Get the active Zotero pane document
let doc = Zotero.getActiveZoteroPane().document;

// Function to click the "Add Tag" button in the active pane
function clickAddTagButtonInActivePane() {
    // Try to find and click the 'Add Tag' button in the main library view
    let activePane = doc.querySelector('vbox#zotero-pane.deck-selected');
    if (activePane) {
        let addButton = activePane.querySelector('tags-box#zotero-editpane-tags toolbarbutton.add.section-custom-button');
        if (addButton) {
            addButton.click();
            return true;
        }
    }

    // Try to find and click the 'Add Tag' button in the PDF view
    activePane = doc.querySelector('deck#zotero-context-pane-item-deck.deck-selected');
    if (activePane) {
        let itemDetails = activePane.querySelector('item-details.zotero-item-pane-content.deck-selected');
        if (itemDetails) {
            let addButton = itemDetails.querySelector('tags-box#zotero-editpane-tags toolbarbutton.add.section-custom-button');
            if (addButton) {
                addButton.click();
                return true;
            }
        }
    }

    // No button found in either view
    return false;
}

// Execute the function
clickAddTagButtonInActivePane();
```

**Comment by mjthoraval** [comment]
Thank you for the script.

The shortcut `Ctrl`+`Shift`+`G` does not work for me in the PDF Reader.
The action works from the Reader actions menu.
It works fine from My Library.

But the shortcut `Shift`+`T` is working fine.

The shortcut also adds a tag to the item when using it on a PDF annotation.
It would be nice if it could add the tag to the annotation instead.

Refs:
https://forums.zotero.org/discussion/117772/add-tag-shortcut
https://forums.zotero.org/discussion/74308/add-tag-shortcut
https://forums.zotero.org/discussion/118402/is-it-possible-to-go-to-the-tags-section-of-each-article-with-a-shortcut

**Comment by neurothew** [comment]
Works like a charm, thanks a lot!

I am using ctrl + t as the shortcut on windows 11.

**Comment by sebadrude** [comment]
I am new to this.  Could you please explain, or point to an explanation, where exactly I am supposed to put this script, and how to connect it to Zotero?  From the Forum https://forums.zotero.org/discussion/117772/add-tag-shortcut I understood that this is related to a plugin called "Actions & Tags", but I am not able to find this plugin in the library of plugins https://www.zotero.org/support/plugins, also not in the unsupported ones.  So how do I use this script/addon, please?

**Reply by windingwind** [reply]
Read the readme pls

**Reply by sebadrude** [reply]
Ah, yes, now I found it: [https://github.com/windingwind/zotero-actions-tags/](https://github.com/windingwind/zotero-actions-tags/)
Thanks!

**Comment by rdiaz02** [comment]
Does this still work with Zotero 8? I had it working just fine up to Zotero 7.0.32, but it seems to have stopped working today, when I upgraded to Zotero 8.

**Reply by powestermark** [reply]
It doesn't work in Zotero 8. A Zotero expert would have to solve it, the UI code has changed too much for me to make the old simple script work. It's unfortunate that keyboard shortcuts seem to not be a priority for the Zotero developers.

**Reply by rdiaz02** [reply]
Thanks a lot for confirming (and, of course, for your original script)! And I agree with you that it is unfortunate that defining and modifying keyboard shortcuts does not seem a priority for Zotero developers.
(For what is worth, I found that https://github.com/windingwind/zotero-actions-tags/discussions/547 still works).

**Reply by powestermark** [reply]
You might appreciate hearing that I solved this with a new Add Tag script for Zotero 8, see #550.

**Reply by rdiaz02** [reply]
Thanks a lot! Using #550 now with Zotero 8 :-)
